src/user/user.service.ts

```diff
@@ -0,0 +1,89 @@
+
+@Injectable()
+export class UserService {
+  constructor(@InjectRepository(User) private userRepository: Repository<User>) {}
+  schemaToDto(schema: User): GetUserDto {
+    //remove sensitive data
+    const dto = { ...schema }
+    delete dto.password
+    return dto
+  }
+  async findOne(
+    conditions: FindConditions<User>,
+    options?: FindOneOptions<User> & { shouldBeElderly?: boolean; shouldExist?: boolean }
+  ): Promise<User> {
+    const user = await this.userRepository.findOne(conditions, options)
+    if (!options) return user
+    if (options.shouldExist && !user) throw new UserNotFoundException()
+    if (
+      options.shouldBeElderly !== undefined &&
+      ((options.shouldBeElderly && !user.isElderly) || (!options.shouldBeElderly && user.isElderly))
+    )
+      throw new InvalidUserTypeException()
+    return user
+  }
+  async createRelationship({ eid, cid }: UpdateRelationshipDto): Promise<User> {
+    //TODO: validate uid from auth
+    const caretaker = await this.findOne({ uid: cid }, { shouldBeElderly: false, shouldExist: true })
+    const elderly = await this.findOne(
+      { uid: eid },
+      { relations: ['takenCareBy'], shouldBeElderly: true, shouldExist: true }
+    )
+    if (elderly.takenCareBy.find((currentCaretaker) => currentCaretaker.uid === cid))
+      throw new DuplicateElementException('Relationship')
+    elderly.takenCareBy.push(caretaker)
+    return await this.userRepository.save(elderly)
+  }
+  async deleteRelationship({ eid, cid }: UpdateRelationshipDto): Promise<User> {
+    //TODO: validate uid from auth
+    await this.findOne({ uid: cid }, { shouldBeElderly: false })
+    const elderly = await this.findOne(
+      { uid: eid },
+      { relations: ['takenCareBy'], shouldBeElderly: true, shouldExist: true }
+    )
+    const newCaretakers = elderly.takenCareBy.filter((c) => c.uid !== cid)
+    return await this.userRepository.save({ uid: elderly.uid, ...elderly, takenCareBy: newCaretakers })
+  }
+  async findCaretakerByElderlyId(uid: number): Promise<User[]> {
+    const elderly = await this.findOne(
+      { uid: uid },
+      { relations: ['takenCareBy'], shouldBeElderly: true, shouldExist: true }
+    )
+    return elderly.takenCareBy
+  }
+  async findElderlyByCaretakerId(uid: number): Promise<User[]> {
+    const caretaker = await this.findOne(
+      { uid: uid },
+      { relations: ['takingCareOf'], shouldBeElderly: false, shouldExist: true }
+    )
+    return caretaker.takingCareOf
+  }
+  async createUser(dto: CreateUserDto): Promise<{ uid: number }[]> {
+    const existingUser = await this.findOne({ phone: dto.phone })
+    if (existingUser) throw new DuplicateElementException('Phone number')
+    const user = this.userRepository.create({
+      ...dto,
+      reminders: [],
+      emotionalRecords: [],
+      memoryPracticeQuestions: [],
+      healthRecords: [],
+      modules: [],
+      takingCareOf: [],
+      takenCareBy: [],
+    })
```

Comment:
`createUser` will either be called by `AuthService` or be moved to `AuthService` (physically moved) since I think hashing and encoding user should be done in `AuthService`
I'm thinking of two approaches
1. Keep `createUser` here and let `AuthService` call this function with hashed password in the input parameter.
2. Move `createUser` to `AuthService`

What do you think?